extras/AFC.py

```diff
@@ -1029,36 +1028,35 @@ def HexConvert(self,tmp):
-                str[UNIT][NAME]["spool_id"]=CUR_LANE.spool_id
-                str[UNIT][NAME]["color"]=CUR_LANE.color
-                str[UNIT][NAME]["weight"]=CUR_LANE.weight
-                str[UNIT][NAME]["runout_lane"]=CUR_LANE.runout_lane
-                filiment_stat=self.get_filament_status(CUR_LANE).split(':')
-                str[UNIT][NAME]['filament_status']=filiment_stat[0]
-                str[UNIT][NAME]['filament_status_led']=filiment_stat[1]
-                str[UNIT][NAME]['status'] = CUR_LANE.status if CUR_LANE.status is not None else ''
-                numoflanes +=1
-            str[UNIT]['system']={}
-            str[UNIT]['system']['type'] = self.printer.lookup_object('AFC_hub '+ UNIT).type
-            str[UNIT]['system']['hub_loaded']  = True == self.printer.lookup_object('AFC_hub '+ UNIT).state
-            str[UNIT]['system']['can_cut']  = True == self.printer.lookup_object('AFC_hub '+ UNIT).cut
-            str[UNIT]['system']['screen'] = screen_mac
+        for UNIT in self.printer.objects:
+            if 'AFC_unit' in UNIT:
+                CUR_UNIT = self.printer.lookup_object(UNIT)
+                str[CUR_UNIT.name]={}
+                for NAME in CUR_UNIT.lanes:
+                    CUR_LANE=self.stepper[NAME]
+                    str[CUR_UNIT.name][CUR_LANE.name]={}
+                    str[CUR_UNIT.name][CUR_LANE.name]['index'] = CUR_LANE.index
+                    str[CUR_UNIT.name][CUR_LANE.name]['hub'] = CUR_LANE.hub
+                    str[CUR_UNIT.name][CUR_LANE.name]['map'] = CUR_LANE.map
+                    str[CUR_UNIT.name][CUR_LANE.name]['load'] = bool(CUR_LANE.load_state)
+                    str[CUR_UNIT.name][CUR_LANE.name]["prep"] =bool(CUR_LANE.prep_state)
+                    str[CUR_UNIT.name][CUR_LANE.name]["tool_loaded"] = CUR_LANE.tool_loaded
+                    str[CUR_UNIT.name][CUR_LANE.name]["loaded_to_hub"] = CUR_LANE.loaded_to_hub
+                    str[CUR_UNIT.name][CUR_LANE.name]["material"]=CUR_LANE.material
+                    str[CUR_UNIT.name][CUR_LANE.name]["spool_id"]=CUR_LANE.spool_id
+                    str[CUR_UNIT.name][CUR_LANE.name]["color"]=CUR_LANE.color
+                    str[CUR_UNIT.name][CUR_LANE.name]["weight"]=CUR_LANE.weight
+                    str[CUR_UNIT.name][CUR_LANE.name]["runout_lane"]=CUR_LANE.runout_lane
+                    filiment_stat=self.get_filament_status(CUR_LANE).split(':')
+                    str[CUR_UNIT.name][CUR_LANE.name]['filament_status']=filiment_stat[0]
+                    str[CUR_UNIT.name][CUR_LANE.name]['filament_status_led']=filiment_stat[1]
+                    str[CUR_UNIT.name][CUR_LANE.name]['status'] = CUR_LANE.status 
+                    numoflanes +=1
+                str[CUR_UNIT.name]['system']={}
+                str[CUR_UNIT.name]['system']['type'] = CUR_UNIT.type
+                #str[UNIT]['system']['hub_loaded']  = True == CUR_UNIT.hub_loaded
```

Comment:
Is this going to be re-added at some point?

Reply:
Has to be moved haven't decided where yet